### PR TITLE
De-dup, cleanup, overhaul networking docs

### PR DESCRIPTION
**Pull request by thockin**
Fix #10312

Not perfect but far less stale.

### CODE REVIEW

File: docs/design/networking.md [hunk 1]
```diff
@@ -1,107 +1,174 @@
...
-
-3. Internet -> Container. This also has to go through the primary host IP and also has 2 levels of NAT, ideally. However, the path currently is a proxy with (External Host IP -> Internal Host IP -> Docker) -> (Docker -> Container IP). Once [issue #15](https://github.com/GoogleCloudPlatform/kubernetes/issues/15) is closed, it should be External Host IP -> Internal Host IP -> Container IP. But to get that second arrow we have to set up the port forwarding iptables rules per mapped port.
-
-Another approach could be to create a new host interface alias for each pod, if we had a way to route an external IP to it. This would eliminate the scheduling constraints resulting from using the host's IP address.
+We want to be able to assign IP addresses externally from Docker ([Docker issue
+#6743](https://github.com/dotcloud/docker/issues/6743)) so that we don't need to
```
Comment (dchen1107) [inline docs/design/networking.md]: copy&paste error here? why make issue number and url so big and change the font color?

File: docs/design/networking.md [hunk 2]
```diff
@@ -1,107 +1,174 @@
...
+statically allocate fixed-size IP ranges to each node, so that IP addresses can
+be made stable across pod infra container restarts ([Docker issue
+#2801](https://github.com/dotcloud/docker/issues/2801)), and to facilitate pod
```
Comment (dchen1107) [inline docs/design/networking.md]: ditto.

**Comment by dchen1107** [comment]
LGTM except nits.

**Comment by thockin** [comment]
linewrap put # at the beginning of the line.    Fixed

**Comment by dchen1107** [comment]
LGTM.

**Comment by yujuhong** [comment]
Merging since this is a docs change.

**Pull request merged**